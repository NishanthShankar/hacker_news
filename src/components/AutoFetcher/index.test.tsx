import Enzyme, { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'
import Component from './index'
Enzyme.configure({ adapter: new Adapter() })

describe('<AutoFetcher />', () => {
  it('renders without props', () => {
    const tree = shallow(<Component />)
    expect(toJson(tree)).toMatchSnapshot()
  })

  it('renders null with hide props', () => {
    const tree = shallow(<Component hide onVisible={() => {}} />)
    expect(toJson(tree)).toMatchSnapshot()
  })
})
