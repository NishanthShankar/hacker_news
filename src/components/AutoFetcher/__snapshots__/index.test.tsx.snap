// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AutoFetcher /> renders null with hide props 1`] = `""`;

exports[`<AutoFetcher /> renders without props 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "height": 100,
      "justifyContent": "center",
      "width": 100,
    }
  }
>
  <Spinner />
</div>
`;
